ext/gd/gd.c

```diff
@@ -4124,9 +4123,8 @@ static void _php_image_output_ctx(INTERNAL_FUNCTION_PARAMETERS, int image_type,
 	char *file = NULL;
 	size_t file_len = 0;
 	zend_long quality, basefilter;
-	zend_bool compressed = 1;
+	zend_bool compressed = 1, quality_is_null = 1, basefilter_is_null = 1;
```

Comment:
Mostly it seems that you are right. However there's https://github.com/php/php-src/blob/2f0dcf2ac2d8459ec1481ca256b74e4959da3979/ext/gd/gd.c#L4232 where I'm not exactly sure what to do. Or just changing the condition to `q != -1` is fine?

Also, there's https://github.com/php/php-src/blob/2f0dcf2ac2d8459ec1481ca256b74e4959da3979/ext/gd/gd.c#L4222 So we could maybe declare a default value of `80` (and change this code) in case of `imagecreatefromwebp`?

Reply:
What about dropping `f` and `q` altogether?